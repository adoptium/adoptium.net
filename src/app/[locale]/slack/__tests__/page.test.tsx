import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import { axe } from "vitest-axe"
import Slack from "../page"

describe("Slack page", () => {
  it("renders correctly", () => {
    const { container } = render(<Slack />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("has no accessibility violations", async () => {
    const { container } = render(<Slack />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
