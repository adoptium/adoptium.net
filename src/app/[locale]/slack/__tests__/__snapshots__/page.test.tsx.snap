// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slack page > renders correctly 1`] = `
<div>
  <div
    class="relative pt-40 pb-16 md:pb-36 md:pt-60 overflow-hidden"
  >
    <div
      class="w-full relative"
    >
      <div
        class="absolute z-[-1] w-[2396px] h-[1340px] left-[23px] md:left-[112px] top-[21px] md:top-[171px] bg-[#410170] shadow-[0_0_400px_rgba(65,1,112,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[1487px] h-[893px] left-[120px] md:left-[676px] top-[120px] md:top-[395px] bg-[#B62987] shadow-[0_0_400px_rgba(182,41,135,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[688px] h-[446px] left-[400px] md:left-[1131px] top-[618px] bg-[#FE8492] shadow-[0_0_400px_rgba(254,132,146,1)] rounded-full blur-[400px]"
      />
    </div>
    <div
      class="mx-auto max-w-[1048px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
    >
      <div
        class="self-stretch flex-col justify-center items-center gap-6 flex"
      >
        <div
          class="self-stretch flex-col justify-center items-center gap-4 flex"
        >
          <div
            class="justify-start items-center gap-3 inline-flex"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FF1464"
                height="8"
                rx="2"
                width="8"
                x="3"
                y="3"
              />
              <rect
                height="11"
                rx="3.5"
                stroke="#FF1464"
                stroke-opacity="0.25"
                stroke-width="3"
                width="11"
                x="1.5"
                y="1.5"
              />
            </svg>
            <div
              class="text-rose-600 text-base font-bold leading-normal"
            >
              Join our Slack community
            </div>
          </div>
          <div
            class="self-stretch text-center text-white text-[56px] lg:text-[80px] leading-[114.286%] md:leading-[120%] font-semibold"
          >
            Slack Signup
          </div>
        </div>
        <div
          class="self-stretch text-center text-lightgrey text-[20px] font-normal leading-[140%] undefined"
        />
      </div>
    </div>
  </div>
  <div
    class="max-w-4xl mx-auto px-4 py-8"
  >
    <p
      class="text-lg mb-4"
    >
      Join our Slack community to connect with other Eclipse Adoptium users and contributors. Share your experiences, ask questions, and collaborate on projects.
    </p>
    <a
      class="inline-block bg-pink text-white px-6 py-3 rounded-lg hover:bg-pink-700 transition-colors"
      href="https://join.slack.com/t/adoptium/shared_invite/zt-37l2atudg-K8SY_QFkmO8zCsgC7MJcWg"
      rel="noopener noreferrer"
      target="_blank"
    >
      Join Slack
    </a>
  </div>
</div>
`;
